src/ast/Event.ts

```diff
@@ -1,27 +1,125 @@
-import DayOfWeek from "./DayOfWeek";
-import Date from "./Date";
-
-class Event extends CalTypes {
-  event: String;
-  calTypes: Array<CalTypes>;
-
-  parse(): void {
-    while (!tokenizer.checkToken(";")) {
-      eventName = tokenizer.getNext();
-      if (tokenizer.checkToken("every") || tokenizer.checkToken("and")) {
-        calTypes.add(new DayOfWeek());
+import Node from "../Node";
+import Tokenizer from "../Tokenizer";
+import { ParserError } from "../errors/ParserError";
+
+export default class Event extends Node {
+  title: string = "";
+  repeating: boolean = false;
+  daysOfWeek: string[] = [];
+  date: string = "";
+  allDay: boolean = false;
+  fromTime: string = "";
+  toTime: string = "";
+  location: string = "";
+  guests: string[] = [];
+
+  parse(tokenizer: Tokenizer): void {
+    // Get what should be the title of the event
+    let currentLine = tokenizer.getLine();
+    let token = tokenizer.pop();
+    if (token === null) {
+      throw new ParserError(`Error on line ${currentLine}: expected an event title`);
+    }
+    this.title = token;
+
+    // Get what should be the date of the event
+    currentLine = tokenizer.getLine();
+    token = tokenizer.pop();
+    if (token === "every") {
+      this.repeating = true;
+      token = tokenizer.pop();
+      if (token === null) {
+        throw new ParserError(`Error on line ${currentLine}: expected a day of the week`);
       }
-      if (tokenizer.checkToken("on")) {
-        calTypes.add(new Date());
+      let dayOfWeek: string = token;
+      this.daysOfWeek.push(dayOfWeek);
+      token = tokenizer.top();
+      // loop over ("and" DAYOFWEEK)* tokens
+      // TODO: clean this up
+      while (token === "and") {
+        tokenizer.pop();
+        currentLine = tokenizer.getLine();
+        token = tokenizer.pop();
+        switch (token) {
+          case "Sunday":
+          case "Monday":
+          case "Tuesday":
+          case "Wednesday":
+          case "Thursday":
+          case "Friday":
+          case "Saturday":
```

Comment:
we could put these into an array and then use `days.contains(token)`

Reply:
Thank you, way better idea than my brain could think of at the moment!